server/lib/chat/chat-service.ts

```diff
@@ -360,6 +367,20 @@ export default class ChatService {
           return assertUnreachable(m.data)
       }
     })
+
+    let mentionIds = Set<SbUserId>()
+    for (const msg of dbMessages) {
+      if (msg.data.type === ServerChatMessageType.TextMessage) {
```

Comment:
imo do this logic when mapping the `TextMessage` above. (It's a bit weird to have an extra side effect in a `map`, but I think it's preferable to iterating the whole list of messages twice)